### PR TITLE
Python packages: major updates

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/jsonrpc-base/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 buildPythonPackage rec {
   pname = "jsonrpc-base";
-  version = "1.0";
+  version = "1.0.1";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1dl55n54ha5kf4x6hap2p1k3s4qa4w7g791wp2656rjg2zxfgywk";
+    sha256 = "21f860c915617f6475aa1ac5a1ec11de03cce6b279741f25ad97d8a4c5b76c3c";
```
Comment (volth) [inline pkgs/development/python-modules/jsonrpc-base/default.nix]: any reason for base32->base16 changes?

Comment (dotlambda) [reply pkgs/development/python-modules/jsonrpc-base/default.nix]: The update script gets the hashes from PyPI, where they are in a base16 format, see https://pypi.org/project/jsonrpc-base/#files.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build conan

File: pkgs/development/tools/build-managers/conan/default.nix [hunk 2]
```diff
@@ -16,16 +16,24 @@ let newPython = python.override {
...
       };
     });
+    astroid = super.astroid.overridePythonAttrs (oldAttrs: rec {
+      version = "1.6.5";
+      disabled = false;
+      src = oldAttrs.src.override {
+        inherit version;
+        sha256 = "fc9b582dba0366e63540982c3944a9230cbc6f303641c51483fa547dcc22393a";
+      };
+    });
```
Comment (dotlambda) [inline pkgs/development/tools/build-managers/conan/default.nix]: @FRidh Do you have any idea why this doesn't work?

> astroid-2.0.1 not supported for interpreter python2.7

Comment (dotlambda) [reply pkgs/development/tools/build-managers/conan/default.nix]: Seems to work when I use `python3`. So, `packageOverrides` must be broken for `python2`.

**Comment by dotlambda** [comment]
@FRidh I'm sorry, I was away for a few days and forgot to revert the spyder changes daabcc3 and 48eb8c4.

**Comment by FRidh** [comment]
@dotlambda actually I removed them and rebased. I don't think the change would have brought us anything, really. Just work for those that had to change their expression.

**Comment by andreabedini** [comment]
This broke `aws-sam-cli` which relies on `aws-sam-translator == 1.6.0` (which this PR updates to 1.6.1). Since I contributed the initial derivation for aws-sam-cli, I'm keen to fix the build but ... how?

**Comment by dotlambda** [comment]
@andreabedini I think the simplest thing is to add a `postPatch` hook, where you use `substituteInPlace` to replace `==1.6.0` by `>=1.6.0`. Plus, you should make upstream aware of it.

**Comment by andreabedini** [comment]
> @andreabedini I think the simplest thing is to add a postPatch hook, where you use substituteInPlace to replace ==1.6.0 by >=1.6.0. Plus, you should make upstream aware of it.

thanks @dotlambda, I will try that

**Pull request merged**